src/backend/metal/src/native.rs

```diff
@@ -81,17 +81,29 @@ impl Default for RasterizerState {
 
 #[derive(Clone, Debug)]
 pub struct DepthStencilState {
-    pub depth_stencil_desc: Option<metal::DepthStencilState>,
+    pub depth_stencil_desc: Option<pso::DepthStencilDesc>,
+    pub depth_stencil_desc_raw: Option<metal::DepthStencilDescriptor>,
+    pub depth_stencil_static: Option<metal::DepthStencilState>,
     pub stencil_front_reference: pso::State<pso::StencilValue>,
```

Comment:
Let's have our `Stencil` struct defined as generic over T being the actual stencil value, and thus we'd be able to re-use it here with `T == pso::State<pso::StencilValue>`

Reply:
I just tried this and it doesn't give us much benefit IMO, except to reduce the repeated `pso::State<pso::StencilValue>` here (but I have to add it a few other places instead). What do you think?